modules/drivers/athena/src/metabase/driver/athena.clj

```diff
@@ -0,0 +1,423 @@
+(defmethod sql.qp/date [:athena :day]     [_driver _unit expr] (hsql/call :date_trunc (hx/literal :day) expr))
+(defmethod sql.qp/date [:athena :month]   [_driver _unit expr] (hsql/call :date_trunc (hx/literal :month) expr))
+(defmethod sql.qp/date [:athena :quarter] [_driver _unit expr] (hsql/call :date_trunc (hx/literal :quarter) expr))
+(defmethod sql.qp/date [:athena :year]    [_driver _unit expr] (hsql/call :date_trunc (hx/literal :year) expr))
+
+(defmethod sql.qp/date [:athena :week]
+  [driver _ expr]
+  (sql.qp/adjust-start-of-week driver (partial hsql/call :date_trunc (hx/literal :week)) expr))
+
+;;;; Datetime extraction functions
+
+(defmethod sql.qp/date [:athena :minute-of-hour]  [_driver _unit expr] (hsql/call :minute expr))
+(defmethod sql.qp/date [:athena :hour-of-day]     [_driver _unit expr] (hsql/call :hour expr))
+(defmethod sql.qp/date [:athena :day-of-month]    [_driver _unit expr] (hsql/call :day_of_month expr))
+(defmethod sql.qp/date [:athena :day-of-year]     [_driver _unit expr] (hsql/call :day_of_year expr))
+(defmethod sql.qp/date [:athena :month-of-year]   [_driver _unit expr] (hsql/call :month expr))
+(defmethod sql.qp/date [:athena :quarter-of-year] [_driver _unit expr] (hsql/call :quarter expr))
+
+(defmethod sql.qp/date [:athena :day-of-week]
+  [driver _ expr]
+  (sql.qp/adjust-day-of-week driver (hsql/call :day_of_week expr)))
+
+(defmethod sql.qp/unix-timestamp->honeysql [:athena :seconds]
+  [_driver _seconds-or-milliseconds expr]
+  (hsql/call :from_unixtime expr))
+
+(defmethod sql.qp/add-interval-honeysql-form :athena
+  [_driver hsql-form amount unit]
+  (hsql/call :date_add
+             (hx/literal (name unit))
+             (hsql/raw (int amount))
+             hsql-form))
+
+(defmethod sql.qp/cast-temporal-string [:athena :Coercion/ISO8601->DateTime]
+  [_driver _semantic-type expr]
+  (hx/->timestamp expr))
+
+(defmethod sql.qp/cast-temporal-string [:athena :Coercion/ISO8601->Date]
+  [_driver _semantic-type expr]
+  (hx/->date expr))
+
+(defmethod sql.qp/cast-temporal-string [:athena :Coercion/ISO8601->Time]
+  [_driver _semantic-type expr]
+  (hx/->time expr))
+
+
+;; fix to allow integer division to be cast as double (float is not supported by athena)
+(defmethod sql.qp/->float :athena
+  [_ value]
+  (hx/cast :double value))
+
+;; Support for median/percentile functions
+(defmethod sql.qp/->honeysql [:athena :median]
+  [driver [_ arg]]
+  (hsql/call :approx_percentile (sql.qp/->honeysql driver arg) 0.5))
+
+(defmethod sql.qp/->honeysql [:athena :percentile]
+  [driver [_ arg p]]
+  (hsql/call :approx_percentile (sql.qp/->honeysql driver arg) (sql.qp/->honeysql driver p)))
+
+(defmethod sql.qp/->honeysql [:athena :regex-match-first]
+  [driver [_ arg pattern]]
+  (hsql/call :regexp_extract (sql.qp/->honeysql driver arg) pattern))
+
+;; keyword function converts database-type variable to a symbol, so we use symbols above to map the types
```

Comment:
I don't know how to parse the sentence "keyword function converts database-type variable to a symbol"

Reply:
Had to poke around to figure this out myself. :) There's a list of mappings from Athena data types to Metabase data types up in `sql-jdbc.sync/database-type->base-type` that uses symbols as it's keys. The database type gets returned as a string so the `keyword` function converts the string database type to a symbol so it can then be used to lookup the proper Metabase type with that function. 😬 

I'm not entirely sure why I added the underlying mapping and didn't just leave the keys as strings and I apologize as I was learning clojure mid-implementation.